### PR TITLE
STL.natvis: add error_code and error_category

### PR DESCRIPTION
**Pull request by vinniefalco**
Thanks for the low-hanging fruit :)

### CODE REVIEW

**Review by StephanTLavavej — CHANGES_REQUESTED** [review]
Thanks for the PR! The checks are failing with:
```
.\stl\debugger\STL.natvis(1,1): error : Validation failed: file contains 6 lines with trailing whitespace.
```

**Comment by vinniefalco** [comment]
You know...I'm just gonna put this out there... I use the Visual Studio IDE editor exclusively... I will update the pull request.

File: stl/debugger/STL.natvis [hunk 1]
```diff
@@ -22,6 +22,20 @@ SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
   </Type>
 
 
+  <Type Name="std::error_category">
+    <DisplayString Condition="_Addr == 1">[future]</DisplayString>
+    <DisplayString Condition="_Addr == 3">[generic]</DisplayString>
+    <DisplayString Condition="_Addr == 5">[iostream]</DisplayString>
+    <DisplayString Condition="_Addr == 7">[system]</DisplayString>
+    <DisplayString>@{((uintptr_t)this),x}</DisplayString>
```
Comment (CaseyCarter) [inline stl/debugger/STL.natvis:30]: I think we should:
```suggestion
    <DisplayString>[{name(),sb}]</DisplayString>
    <Expand>
      <Synthetic Name="hint">
        <DisplayString>Enable "Allow Function Calls In Value Formatting" if you see "???" here</DisplayString>
      </Synthetic>
    </Expand>
```
here. It requires "Allow Function Calls In Value Formatting" to be enabled in the debugger, but produces an identical experience for library-defined and user-defined `error_category`s.

File: stl/debugger/STL.natvis [hunk 2]
```diff
@@ -22,6 +22,20 @@ SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
...
+    <DisplayString Condition="_Addr == 1">[future]</DisplayString>
+    <DisplayString Condition="_Addr == 3">[generic]</DisplayString>
+    <DisplayString Condition="_Addr == 5">[iostream]</DisplayString>
+    <DisplayString Condition="_Addr == 7">[system]</DisplayString>
+    <DisplayString>@{((uintptr_t)this),x}</DisplayString>
+  </Type>
+
+
+  <Type Name="std::error_code">
+    <DisplayString>{_Myval} {*_Mycat}</DisplayString>
```
Comment (CaseyCarter) [inline stl/debugger/STL.natvis:35]: If we accept the above suggestion, we should also:
```suggestion
    <DisplayString>{_Myval} {*_Mycat}</DisplayString>
    <Expand>
      <Synthetic Name="hint">
        <DisplayString>Enable "Allow Function Calls In Value Formatting" if you see "???" for the category</DisplayString>
      </Synthetic>
    </Expand>
```
for discoverability.

**Comment by CaseyCarter** [comment]
For other reviewers: Paste this program
```c++
#include <future>
#include <ios>
#include <string>
#include <system_error>

struct my_error_category : public std::error_category {
	const char* name() const noexcept override {
		return "potato";
	}

	std::string message(int) const override {
		return "message goes here";
	}
};

int main() {
	auto& generic_cat = std::generic_category();
	auto& system_cat = std::system_category();
	auto& iostream_cat = std::iostream_category();
	auto& future_cat = std::future_category();
	const auto& my_cat = my_error_category();

	std::error_code generic(42, generic_cat);
	std::error_code system(42, system_cat);
	std::error_code iostream(42, iostream_cat);
	std::error_code future(42, future_cat);
	std::error_code my(42, my_cat);
}
```
into a VS project. Add a new natvis file with the new visualizers from the PR. Set a breakpoint at the final `}`, and let it rip.

**Review by CaseyCarter — CHANGES_REQUESTED** [review]

**Comment by StephanTLavavej** [comment]
This visualizer needs to be updated now that #3139 has been merged.

**Comment by StephanTLavavej** [comment]
Closing as this has been superseded by #3204. Thanks again for looking into this!

**Pull request closed without merging**